x/ecocredit/client/tx.go

```diff
@@ -504,3 +498,66 @@ Parameters:
 		},
 	})
 }
+
+// TxCreateProject returns a transaction command that creates a new project.
+func TxCreateProject() *cobra.Command {
+	cmd := txflags(&cobra.Command{
+		Use:   "create-project [class-id] [project-location] [metadata] --project-id [project-id]",
+		Short: "Create a new project within a credit class",
+		Long: `Create a new project within a credit class.
+		
+		Parameters:
+		class-id: id of the class
+		project-location: the location of the project (see documentation for proper project-location formats).
+		metadata: base64 encoded metadata - any arbitrary metadata attached to the project.
+		project-id: id of the project (optional).
+		`,
+		Args: cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if args[0] == "" {
+				return errors.New("class-id is required")
+			}
+
+			classID := args[0]
+
+			if args[1] == "" {
+				return errors.New("project location is required")
+			}
+			projectLocation := args[1]
```

Comment:
let's validate the location here. No need to create a transaction if the project location is not valid. We should create a function and export it and use it in both Msg server and here for validation.